src/core/DY.Core.Bytes.fst

```diff
@@ -334,55 +362,78 @@ let get_dh_usage #cusages pk =
 /// by honest principals.
 
 noeq
-type crypto_predicates (cusages:crypto_usages) = {
-  aead_pred: tr:trace -> key:bytes{AeadKey? (get_usage key)} -> nonce:bytes -> msg:bytes -> ad:bytes -> prop;
-  aead_pred_later:
+type aead_crypto_predicate (cusages:crypto_usages) = {
+  pred: tr:trace -> key:bytes{AeadKey? (get_usage key)} -> msg:bytes -> ad:bytes -> prop;
+  pred_later:
     tr1:trace -> tr2:trace ->
-    key:bytes{AeadKey? (get_usage key)} -> nonce:bytes -> msg:bytes -> ad:bytes ->
+    key:bytes{AeadKey? (get_usage key)} -> msg:bytes -> ad:bytes ->
```

Comment:
what happend to the `nonce` argument?

Reply:
I thought it was not authenticated by the AEAD so I removed it, but after more thorough researches it look like it is authenticated! I will add it back to the predicate.